src/chips/sam4l/dma.rs

```diff
@@ -212,6 +216,25 @@ impl DMAChannel {
 
         volatile_store(&mut registers.interrupt_enable, 1 << 1);
     }
+
+    pub fn do_xfer_buf(&self, pid: usize, 
+                       buf: Option<&'static mut [u8]>,
+                       len: usize) {
+        let ptr = buf.unwrap();
+        if len > ptr.len() {
```

Comment:
I agree. Let's keep that in mind. I'm wary of pre-design, designing before you really know the requirements. I.e., we know we'll want to design this, but since we don't yet have enough experience can't with a lot of confidence know what its requirements are.